stdlib/bytes.mli

```diff
@@ -173,6 +173,22 @@ val mapi : (int -> char -> char) -> bytes -> bytes
     index (in increasing index order) and stores the resulting bytes
     in a new sequence that is returned as the result. *)
 
+val fold : (char -> 'a -> 'a) -> bytes -> 'a -> 'a
```

Comment:
One should try to mimic the interface of `Array` (since `bytes` is really just a compact form of `char array`), i.e. provide both `fold_left` and `fold_right`.